adagrams/game.py

```diff
@@ -1,11 +1,129 @@
+        letter_list.extend([letter] * count)
 
+    # Draw a random sample of 10 letters from the letter list
+    hand = random.sample(letter_list, 10)
+    
+    return hand
+
+# Function to check if the given word can be formed using the available letters
 def uses_available_letters(word, letter_bank):
-    pass
+    word = word.upper()
+        # Create a copy of the letter bank to avoid modifying the original
+
+    letter_bank_copy = letter_bank.copy()
+
+    # Iterate through each letter in the word
+    for letter in word:
+        if letter in letter_bank_copy:
+            letter_bank_copy.remove(letter)
+        else:
+            return False
+
+    # If all letters are in the letter bank copy, the word is valid
+    return True
 
 def score_word(word):
-    pass
+    word = word.upper()
+    letter_scores = {
+        "A": 1, "E": 1, "I": 1, "O": 1, "U": 1, "L": 1, "N": 1, "R": 1, "S": 1, "T": 1,
+        "D": 2, "G": 2,
+        "B": 3, "C": 3, "M": 3, "P": 3,
+        "F": 4, "H": 4, "V": 4, "W": 4, "Y": 4,
+        "K": 5,
+        "J": 8, "X": 8,
+        "Q": 10, "Z": 10
+    }
+
+    score = 0
+    for letter in word:
+        score += letter_scores[letter]
+
+    if 7 <= len(word) <= 10:
+        score += 8
+
+    return score
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    # Calculate the score for each word in the word_list and store them as tuples
+    word_scores = [(word, score_word(word)) for word in word_list]
+    
+    # Initialize the highest_word variable with the first word and its score
+    highest_word = word_scores[0]
+
+    # Iterate through the rest of the word_scores list (skipping the first element)
+    for word, score in word_scores[1:]:
+        # Check if the current word has a higher score than the highest_word
+        if score > highest_word[1] or (
+            # If the scores are equal, apply tie-breaking rules:
+            score == highest_word[1] and (
+                # If the current word has 10 letters and the highest_word doesn't, choose the current word
+                (len(word) == 10 and len(highest_word[0]) != 10) or
+                # If neither word has 10 letters, choose the word with fewer letters
+                (len(highest_word[0]) != 10 and len(word) < len(highest_word[0]))
+            )
+        ):
+            # Update the highest_word with the current word and its score
+                    highest_word = (word, score)
+        # If the scores are equal and the word lengths are also equal, use the order in the supplied list
+        elif score == highest_word[1] and len(word) == len(highest_word[0]):
+            # Check the index of the current word and highest_word in the original list
+            # If the index of the current word is lower, update highest_word with the current word and its score
+            highest_word = (word_list.index(word) < word_list.index(highest_word[0])) and (word, score) or highest_word
```

Comment:
These statements are a bit hard to read. For the sake of keeping the code readable, it might be worth it to have if and elif statements here to lay out the logic rather than trying to get everything together with `and`s and `or`s.